openmicroscopy/openmicroscopy
# Get channels no re

Resolves a long-standing issue of unnecessarily loading the Rendering Engine when displaying the right "General" panel.

Blitz Gateway image.getChannels() loads the RE by default, allowing channel.getColor() etc. but this is not needed for the General panel.
A new method, image.getChannelsNoRE() does not attempt to load the rendering engine and this is now used for the General pane.

To test:
 - Import a big image (jpeg/png) and try loading General panel during Pyramid generation (when rendering engine throws MissingPyramidException. Panel should load without exception and show channel names.
 - Test that editing channel names works.
 - Also try viewing channel names for SPW images.
